### PR TITLE
VCI-100MKII: improve fx super/mix knob and more

### PR DESCRIPTION
**Pull request by sohet**
Two minor fixes.
- avoid slip of fx super/mix knob (obsolete after the recent commit 10/24)

When using one knob for fx super1 (without shift) and wet/dry mix (with shift) with soft takeover, slips at the edge (max or min position) occur in some cases. For example,
1. super1 is 0.0, mix is 1.0 and the knob is at min.
2. turn the knob to max without shift --> super1 is 1.0, mix is 1.0
3. turn the knob left with shift to diminish mix --> slip! fail to diminish mix, need back and forth again

The slip can be avoided when turning the knob extremely slow, because there is no midi value leap then. But I want to avoid it always, so I set the same value again on key change from super1 to mix (and mix to super1). Other controllers may need the same tweak.
- avoid possible conflicts between scratch and brake

Brake process is stopped when the wheel is released even without shift key.

### CODE REVIEW

File: res/controllers/Vestax-VCI-100MKII-scripts.js [hunk 1]
```diff
@@ -171,14 +171,18 @@ VCI102.pitch = function(ch, midino, value, status, group) {
...
-        engine.setValue(group, "super1", value / 127);
-        engine.softTakeoverIgnoreNextValue(group, "mix");
+    var key = VCI102.shift[ch % 2] ? "mix" : "super1";
+    if (VCI102.superKnobKey[ch] != key) {
+        engine.softTakeoverIgnoreNextValue(group, key);
+        // set previous value again on key change to avoid slip
+        engine.setValue(group, key, VCI102.superKnobValue[ch]);
+        VCI102.superKnobKey[ch] = key;
     }
+    engine.setValue(group, key, VCI102.superKnobValue[ch] = value / 127);
```
Comment (Be-ing) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: IMO it would be more readable to not set a variable while passing a parameter to a function.

Comment (sohet) [inline res/controllers/Vestax-VCI-100MKII-scripts.js]: OK! I'll divide the line.

**Comment by sohet** [comment]
I have no plans to revise this mapping these days. Please merge it as is.

**Comment by sohet** [comment]
* change filter mapping (set default to low-high pass and enable customization in the js file)
* exchange pregain and pitch knobs (a la Deer skin)

**Comment by sohet** [comment]
Add control of Effect2 of the EffectUnit using shift of the other deck (this controller has two shift buttons emitting different midi codes).

http://www.mixxx.org/wiki/doku.php/vestax_vci-100mkii#mapping_for_mixxx_development

**Comment by sohet** [comment]
It is annoying for me to control the super/mix(with shift) effect knob even with careful soft takeover. So I change the behavior of the knob, and drop shift control and soft takeover.

* Assign super1 to the knob if any parameters are linked to it
* Assign mix to the knob if no parameter is linked to it

**Comment by Be-ing** [comment]
The changes LGTM. I have a suggestion for a different way to map the knobs. This only makes sense if you don't really use the key knob. IMO adjusting the key of a track usually sounds odd so I never use it myself.

Corresponding to the knobs as labeled on the wiki:
6: superknob
11: dry/wet
10: filter/Quick Effect superknob
12: gain

I remember you saying before that the gray knob 11 doesn't have a center detent, so mapping both sides of a filter was hard to use. So, I think it would be good to move the filter knob to a knob that does have a center detent and use the gray knob for something that a center detent wouldn't be helpful.

**Comment by sohet** [comment]
Thank you again for your review and suggestion.

> This only makes sense if you don't really use the key knob. IMO adjusting the key of a track usually sounds odd so I never use it myself.

I really use it! I also don't use pitch change in most cases, but when the key difference of tracks is 1-semitone or 6-semitones, I want to use it especially for long term layering. Even in this case, 1-semitone up or down sounds enough, and this small pitch change does not degenerate the sound so much, I think. More distant key change (maybe 3-semitones is enough) and sync_key (helpfull for those not being sure about key theory) are optional.

> I remember you saying before that the gray knob 11 doesn't have a center detent, so mapping both sides of a filter was hard to use. So, I think it would be good to move the filter knob to a knob that does have a center detent and use the gray knob for something which a center detent wouldn't be helpful for.

I have tried many times and many ways for low-high, but this only-high-pass has been always better for this old controller. Any way, ready for merge. I'll try your new library when I map another controller!

**Comment by Be-ing** [comment]
Alrighty, LGTM! 👍 

If you could look through the library as it is now and think if there's anything missing that would be helpful for this controller, that would be appreciated. Of course the big thing missing right now is a JogWheel object. That will come...

**Comment by Be-ing** [comment]
You may be interested in looking at how I used the library to map effects for the P32:https://github.com/mixxxdj/mixxx/pull/1054/files#diff-400922cb0109e689c3d5404810aec4dfR1065

**Pull request merged**